skimage/filters/_gaussian.py

```diff
@@ -112,9 +112,17 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
             sigma = [sigma] * (image.ndim - 1)
         if len(sigma) != image.ndim:
             sigma = np.concatenate((np.asarray(sigma), [0]))
-    image = convert_to_float(image, preserve_range)
-    return ndi.gaussian_filter(image, sigma, mode=mode, cval=cval,
-                               truncate=truncate)
+    if output is not None:
+        if not isinstance(output, np.ndarray):
+            raise ValueError("Output is not an numpy array")
+        else:
```

Comment:
I think that you can remove this `else` block: what ever is image dtype, you can call `convert_to_float` an then check if output dtype corresponds to the converted image dtype and warn the user if it is not the case.